v2/tools/generator/internal/reporting/testdata/Test_TypeCatalogReport_GivenInterface_ShowsExpectedDetails.golden

```diff
@@ -0,0 +1,2 @@
+github.com/Azure/azure-service-operator/testing/person/v20200101
+└── Person: Interface
```

Comment:
Does your test case include a function? Maybe it should include a couple so we can see the sorting works?

Reply:
I just found the test - it doesn't. Suggest adding a couple.

Reply:
It doesn't but the default is also to not show functions, right? I used the same "should I show functions" option as ObjectType which I think we default to false (don't show functions)?

I can write a test that tests the function showing but wasn't sure if it made sense given that's not what the default behavior is and so isn't what people are going to see in the generated files.

Reply:
The TypeCatalog is used in a couple of places, one of which shows functions, one of which doesn't.

Using the same behaviour as ObjectType is good - but since it's a separate test, wouldn't it be easy to show functions from `Test_TypeCatalogReport_GivenInterface_ShowsExpectedDetails`? We already have `test.NewFakeFunction()` which can be used.

Reply:
You're right that we're missing this test for `ObjectType` - I'll add one after this PR merges.